src/main/kotlin/uk/gov/justice/digital/hmpps/personrecord/message/processors/delius/OffenderEventProcessor.kt

```diff
@@ -4,47 +4,42 @@ import kotlinx.coroutines.runBlocking
 import org.slf4j.LoggerFactory
 import org.springframework.beans.factory.annotation.Value
 import org.springframework.stereotype.Component
-import uk.gov.justice.digital.hmpps.personrecord.client.OffenderDetailRestClient
-import uk.gov.justice.digital.hmpps.personrecord.client.model.offender.DeliusOffenderDetail
+import uk.gov.justice.digital.hmpps.personrecord.client.CorePersonRecordAndDeliusClient
+import uk.gov.justice.digital.hmpps.personrecord.client.model.offender.ProbationCase
 import uk.gov.justice.digital.hmpps.personrecord.jpa.repository.PersonRepository
-import uk.gov.justice.digital.hmpps.personrecord.message.processors.EventProcessor
 import uk.gov.justice.digital.hmpps.personrecord.model.DomainEvent
 import uk.gov.justice.digital.hmpps.personrecord.model.person.Person
 import uk.gov.justice.digital.hmpps.personrecord.service.PersonService
 import uk.gov.justice.digital.hmpps.personrecord.service.RetryExecutor
 import uk.gov.justice.digital.hmpps.personrecord.service.TelemetryService
 import uk.gov.justice.digital.hmpps.personrecord.service.type.TelemetryEventType.DELIUS_RECORD_CREATION_RECEIVED
-import java.net.URI
 
-const val NEW_OFFENDER_CREATED = "probation-case.engagement.created"
 const val MAX_RETRY_ATTEMPTS: Int = 3
 
-@Component(value = NEW_OFFENDER_CREATED)
-class OffenderCreatedEventProcessor(
+@Component
+class OffenderEventProcessor(
   val telemetryService: TelemetryService,
-  val offenderDetailRestClient: OffenderDetailRestClient,
+  val corePersonRecordAndDeliusClient: CorePersonRecordAndDeliusClient,
   val personService: PersonService,
   val personRepository: PersonRepository,
-) : EventProcessor() {
+) {
   companion object {
     @Value("\${retry.delay}")
     private val retryDelay: Long = 0
     private val log = LoggerFactory.getLogger(this::class.java)
   }
 
-  override fun processEvent(domainEvent: DomainEvent) {
-    val offenderDetailUrl = domainEvent.detailUrl
-    val crnIdentifier = domainEvent.personReference?.identifiers?.first { it.type == "CRN" }
+  fun processEvent(domainEvent: DomainEvent) {
+    val crn = domainEvent.personReference?.identifiers?.first { it.type == "CRN" }!!.value
     telemetryService.trackEvent(
       DELIUS_RECORD_CREATION_RECEIVED,
```

Comment:
I know this is not correct any more, there is a meeting at 10 to try to decide what to do.
I quite like `DOMAIN_EVENT_RECEIVED` with `TYPE: OFFENDER_ADDRESS_CHANGED`